ee/presencedetection/presencedetection_windows.go

```diff
@@ -0,0 +1,178 @@
+	resultNotConfiguredForUser uintptr = 0x2
+	resultDisabledByPolicy     uintptr = 0x3
+	resultDeviceBusy           uintptr = 0x4
+	resultRetriesExhausted     uintptr = 0x5
+	resultCanceled             uintptr = 0x6
+)
+
+var resultErrorMessageMap = map[uintptr]string{
+	resultDeviceNotPresent:     "There is no authentication device available.",
+	resultNotConfiguredForUser: "An authentication verifier device is not configured for this user.",
+	resultDisabledByPolicy:     "Group policy has disabled authentication device verification.",
+	resultDeviceBusy:           "The authentication device is performing an operation and is unavailable.",
+	resultRetriesExhausted:     "After 10 attempts, the original verification request and all subsequent attempts at the same verification were not verified.",
+	resultCanceled:             "The verification operation was canceled.",
+}
+
+// IUserConsentVerifierInterop is the interop interface for UserConsentVerifier. Both are documented here:
+// https://learn.microsoft.com/en-us/uwp/api/windows.security.credentials.ui.userconsentverifier?view=winrt-26100#desktop-apps-using-cwinrt
+type IUserConsentVerifierInterop struct {
+	ole.IInspectable
+}
+
+func (v *IUserConsentVerifierInterop) VTable() *IUserConsentVerifierInteropVTable {
+	return (*IUserConsentVerifierInteropVTable)(unsafe.Pointer(v.RawVTable))
+}
+
+type IUserConsentVerifierInteropVTable struct {
+	ole.IInspectableVtbl
+	RequestVerificationForWindowAsync uintptr
+}
+
+var roInitialize = sync.OnceFunc(func() {
+	// Call ole.RoInitialize(1) only once
+	ole.RoInitialize(1)
+})
+
+// Detect prompts the user via Hello.
+func Detect(reason string) (bool, error) {
+	roInitialize()
+
+	if err := requestVerification(reason); err != nil {
+		return false, fmt.Errorf("requesting verification: %w", err)
+	}
+
+	return true, nil
+}
+
+// requestVerification calls Windows.Security.Credentials.UI.UserConsentVerifier.RequestVerificationAsync via the interop interface.
+// See: https://learn.microsoft.com/en-us/windows/win32/api/userconsentverifierinterop/nf-userconsentverifierinterop-iuserconsentverifierinterop-requestverificationforwindowasync
+func requestVerification(reason string) error {
+	// Get access to UserConsentVerifier via factory
+	factory, err := ole.RoGetActivationFactory("Windows.Security.Credentials.UI.UserConsentVerifier", iUserConsentVerifierStaticsGuid)
+	if err != nil {
+		return fmt.Errorf("getting activation factory for UserConsentVerifier: %w", err)
+	}
+	defer factory.Release()
+
+	// Query for the interop interface, which we need to actually interact with this method
+	verifierObj, err := factory.QueryInterface(iUserConsentVerifierInteropGuid)
+	if err != nil {
+		return fmt.Errorf("getting UserConsentVerifier from factory: %w", err)
+	}
+	defer verifierObj.Release()
+	verifier := (*IUserConsentVerifierInterop)(unsafe.Pointer(verifierObj))
+
+	// Get the current window handle from systray
+	windowHwnd, err := systray.WindowHandle()
```

Comment:
```suggestion
	windowHandler, err := systray.WindowHandle()
```
I think? ... just a nit

Reply:
`HWND` is the Windows type for a window handle -- `windowHwnd` (`window handle [to] window`) is redundant, though 😅 . I'll update!